src/jdk.compiler/share/classes/com/sun/tools/javac/tree/TreeInfo.java

```diff
@@ -276,6 +276,27 @@ public static boolean isDiamond(JCTree tree) {
         }
     }
 
+    /** Return true if a default expression may represent a poly expression (with elided types) */
+    public static boolean isPossiblePolyDefault(JCDefaultValue tree, Type clazztype) {
+        JCTypeApply applyTree = TreeInfo.getTypeApplication(tree.clazz);
+        if (applyTree != null) {
+            return applyTree.arguments.isEmpty();
+        } else {
+            // No type arguments before .default - Consider if the type is generic or not
+            return clazztype == null || clazztype.tsym.type.isParameterized();
+        }
```

Comment:
This method is problematic - This will always treat Foo.default as a poly expression in an invocation context even when the concerned primitive type is not a generic class.  For instance the Foo.default expression in the following snippet becomes a poly expression when it is a standalone expression in reality:

```
primitive class X {

    static void foo(Object o) {
    }

    public static void main(String [] args) {
        foo(X.default);    
    }

}

```